File: Iniciante/2221/2221.c
#include <stdio.h>

int main() {
	int T, B;
	int A[2], D[2], L[2], Golpe[2];
	int i;
	
	scanf("%d",&T);
	
	for(i=1; i<=T; i++){
		scanf("%d",&B);
		scanf("%d %d %d",&A[0], &D[0], &L[0]);
		scanf("%d %d %d",&A[1], &D[1], &L[1]);
		
		Golpe[0] = (A[0] + D[0])/2;
		Golpe[1] = (A[1] + D[1])/2;
		
		if(L[0]%2 == 0)
			Golpe[0] = Golpe[0] + B;
			
		if(L[1]%2 == 0)
			Golpe[1] = Golpe[1] + B;
			
		if(Golpe[0] == Golpe[1])
			printf("Empate\n");
		
		else{
			
			if(Golpe[0] > Golpe[1])
				printf("Dabriel\n");
			else
				printf("Guarte\n");
				
		}
	}
	
	return 0;
}

File: Python/Iniciante/2221.py
# Problema 2221 - Beecrowd - Iniciante - Nível 2

#Função para fazer o cálculo do valor do golpe 
def golpe(x, bonus):
    ValorGolpe = float((x[0]+x[1])/2)
    if (x[2] % 2) == 0:
        ValorGolpe += bonus
    return ValorGolpe

#Recebe quantas vezes o código vai rodar
instancias = int(input())

#Cria um loop que decrementa a quantidade de instancias até chegar em zero
while instancias:
    instancias -= 1

    #Anota o buff que será utilizado no cálculo
    bonus = int(input())

    #Recebe os dados dos jogadores
    DadosDabriel = list(map(int,input().split()))
    DadosGuarte = list(map(int,input().split()))

    #Chama a função para fazer o cálculo
    ValorDabriel = golpe(DadosDabriel, bonus)
    ValorGuarte = golpe(DadosGuarte, bonus)

    #Compara o valor obtido para ver qual foi o ganhador
    if ValorDabriel > ValorGuarte:
        print('Dabriel')
    elif ValorGuarte > ValorDabriel:
        print('Guarte')
    else:
        print('Empate')
